.github/workflows/ci.yml

```diff
@@ -69,8 +69,18 @@ jobs:
       - name: Setup OS matrix
         id: setup-os-matrix
         run: |
-          os='["ubuntu-latest","windows-latest","macos-latest"'
+          os='["ubuntu-latest","windows-latest"'
+
+          # runs on macOS only if there is a push to master, or a tag is 
+          # pushed, we do this since macOS builds last too long and ILGPU 
+          # is rarely used on a macOS
+          [ "${{ github.ref }}" == "refs/heads/master" ] ||
+          [[ "${{ github.ref }}" =~ "refs/tags/v" ]] && 
+          [ "${{ github.event_name }}" == "push" ] &&
```

Comment:
is this necessary?

Reply:
i think yes, since we also want to avoid `pull_request` events, so I guess it can stay

Reply:
good point @ljubon - we should actually do the same in the other conditions about pushes to master/tag, i.e. the publish/release jobs (talking about GHA conditions here, not shell)

Reply:
also I would reorganise and group the conditions to something like:
```bash
          [ "${{ github.event_name }}" == "push" ] &&
          ([ "${{ github.ref }}" == "refs/heads/master" ] ||
          [[ "${{ github.ref }}" =~ "refs/tags/v" ]])
```

Reply:
also also we might want to switch to using https://github.com/actions/github-script so we can do proper JSON processing (I know no one here likes JavaScript 😅) OR we could use PowerShell and its `ConvertTo-Json -Compress` cmdlet (that might fit this project better 😉)

Reply:
json would naturally be easiest with the java script, so it would be my pick(even tho I am not a huge fan :D)
tho if we are going for a shell I would prefer bash over powerShell, i believe its a more common shell between the two, with `jq` its something like:
```
OSES="[]"
OSES=$(echo "example-os" | jq -R --argjson oses "$OSES" '$oses + [.]')
```

Reply:
I really like your second proposal! ⚡
Maybe with variable name `os` instead of `_OSES` for readability?

Something like:
```bash
os=("os1" "os2")
os+=("os3")
echo "::set-output name=os::$(jq -cn '$ARGS.positional' --args ${os[@]})"
```